proxy/src/main/java/com/wavefront/agent/PushAgent.java

```diff
@@ -325,16 +326,34 @@ protected void startListeners() {
 
     portIterator(traceListenerPorts).forEachRemaining(strPort ->
         startTraceListener(strPort, handlerFactory, compositeSampler));
-    portIterator(traceJaegerListenerPorts).forEachRemaining(strPort -> {
-      PreprocessorRuleMetrics ruleMetrics = new PreprocessorRuleMetrics(
-          Metrics.newCounter(new TaggedMetricName("point.spanSanitize", "count", "port", strPort)),
-          null, null
-      );
-      preprocessors.getSystemPreprocessor(strPort).forSpan().addTransformer(
-          new SpanSanitizeTransformer(ruleMetrics));
-      startTraceJaegerListener(strPort, handlerFactory,
-          new InternalProxyWavefrontClient(handlerFactory, strPort), compositeSampler);
-    });
+
+    if (traceJaegerListenerProtocol.equals("tchannel")) {
+      portIterator(traceJaegerListenerPorts).forEachRemaining(strPort -> {
+        PreprocessorRuleMetrics ruleMetrics = new PreprocessorRuleMetrics(
+            Metrics.newCounter(new TaggedMetricName("point.spanSanitize", "count", "port", strPort)),
+            null, null
+        );
+        preprocessors.getSystemPreprocessor(strPort).forSpan().addTransformer(
+            new SpanSanitizeTransformer(ruleMetrics));
+        startTraceJaegerListener(strPort, handlerFactory,
+            new InternalProxyWavefrontClient(handlerFactory, strPort), compositeSampler);
+      });
+    } else if (traceJaegerListenerProtocol.equals("http")) {
+      portIterator(traceJaegerListenerPorts).forEachRemaining(strPort -> {
+        PreprocessorRuleMetrics ruleMetrics = new PreprocessorRuleMetrics(
+            Metrics.newCounter(new TaggedMetricName("point.spanSanitize", "count", "port", strPort)),
+            null, null
+        );
+        preprocessors.getSystemPreprocessor(strPort).forSpan().addTransformer(
+            new SpanSanitizeTransformer(ruleMetrics));
+        startTraceJaegerHttpListener(strPort, handlerFactory,
+            new InternalProxyWavefrontClient(handlerFactory, strPort), compositeSampler);
+      });
+    } else {
+      logger.severe("Unable to start Jaeger listener due to invalid listener protocol: " +
```

Comment:
should we just log a warn message and default to use jaeger with tchannel listener instead in this case ?

Reply:
no longer applicable